### PR TITLE
[script.skin.info.service] v1.0.3

### PR DESCRIPTION
**Pull request by mikesilvo164**
### Add-on details:

- **General**
  - Add-on name: Skin Info Service
  - Add-on ID: script.skin.info.service
  - Version number: 1.0.3
  - Kodi/repository version: Leia

- **Code location**
  - URL: https://github.com/mikesilvo164/script.skin.info.service
  - Revision/tag: ae0e54f
  - Branch: master

### Description of changes:

Script used by skins to show extended media info.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.skin.info.service/Utils.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            return html
+        except:
+            log("GetStringFromURL: could not get data from %s" % encurl)
+            xbmc.sleep(1000)
+            succeed += 1
+    return ""
+
+
+def Notify(header, line='', line2='', line3=''):
+    xbmc.executebuiltin('Notification(%s, %s, %s, %s)' % (header, line, line2, line3))
```
Comment (Rechi) [inline script.skin.info.service/Utils.py]: Use `xbmcgui.Dialog().notification(...)`.

File: script.skin.info.service/addon.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.skin.info.service" name="Skin Info Service" version="1.0.3" provider-name="phil65">
```
Comment (Rechi) [inline script.skin.info.service/addon.xml]: Add yourself to the provider-name list.

**Comment by mikesilvo164** [comment]
@Rechi @ronie I am trying to wrap my head around making the script python 3 compatible and have read the wiki and other sources but I am currently learning Python as I go here so it is taking me some time to grasp all of this.  I am having some trouble figuring out what other changes should be done.

Any help or advice would be much appreciated.

Thanks 😃

**Comment by ronie** [comment]
there's two paths you can follow:
- modify the addon to be compatible with both py2 and py3
- create two versions of the addon, one compatible with py2 (the current one) and one compatible with py3

we provide kodi py3 builds for testing, they can be found on our mirrors: http://mirrors.kodi.tv/test-builds/
just download and install one of the 'feature_python3' build for your platform.

now test your addon using the 'feature_python3' build and keep an eye on the debuglog for errors/crashes. fix the error in your addon and test again. rinse and repeat until it runs smoothly.

**Comment by enen92** [comment]
ping @mikesilvo164

**Comment by mikesilvo164** [comment]
Hi @enen92 @ronie 

It took me awhile to figure some stuff out here and I have tested using the KodiSetup-20190728-82ed8e1d-PR16116-merge-x64.exe test build and the current KodiSetup-20190729-55d75d97-master-x64.exe for Matrix.

All of these changes were one at a time using the errors in the log file to figure out where the problems are and I think I have it working now.

Not sure about how I handled some of the version checking so any suggestions would be most welcome.

Thanks 😃

**Comment by mikesilvo164** [comment]
Hi @Rechi @ronie @enen92

I noticed that this script currently does not prefix the window properties with anything so I have a pending update to this PR to add SkinInfo as a prefix:

Example: **Window(Home).Property(SkinInfo.Set.Movies.Director)**

Instead of the current:

**Window(Home).Property(Set.Movies.Director)**

I have that ready to go and a skin update is also pending whether or not this can get in the Kodi repo cause I would like to update my skin at the same time this goes in.

I haven't pushed anything yet cause I did not want this to accidentally go in before my skin was ready so I just wanted to bring this up for when anyone has some time to look this over.

Please let me know if there are any other changes required to get this in the repo.

Thanks.

**Comment by enen92** [comment]
@mikesilvo164 just close it and reopen it when it is ready

**Comment by mikesilvo164** [comment]
@enen92 I have both the skin and add-on ready to go.
[https://github.com/xbmc/repo-skins/issues/249](https://github.com/xbmc/repo-skins/issues/249)

😃

File: script.skin.info.service/addon.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+<addon id="script.skin.info.service" name="Skin Info Service" version="1.0.4" provider-name="phil65, SiLVO">
+	<requires>
+		<import addon="xbmc.python" version="2.26.0" />
+	</requires>
+	<extension point="xbmc.python.library"          library="default.py" />
+	<extension point="xbmc.python.script"           library="default.py">
+		<provides>executable</provides>
+	</extension>
+	<extension point="xbmc.addon.metadata">
+		<summary lang="en">Script used by skins to show extended media info.</summary>
```
Comment (enen92) [inline script.skin.info.service/addon.xml]: en_GB please

File: script.skin.info.service/addon.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		<description lang="en">This script fetches extended information for several media types (artists, albums, movie sets etc.). Needs skin support.</description>
```
Comment (enen92) [inline script.skin.info.service/addon.xml]: en_GB please

**Comment by mikesilvo164** [comment]
@enen92 Changes done.

**Comment by enen92** [comment]
Merged

**Comment by mikesilvo164** [comment]
@enen92 Thanks. Any idea when the skin can get updated in the Kodi repo as well?

**Comment by enen92** [comment]
I don't review skins, you'll have to wait for some skinner to have a look into it.

**Pull request merged**